src/common/snippets/src/op/load.cpp

```diff
@@ -29,13 +32,11 @@ std::shared_ptr<Node> Load::clone_with_new_inputs(const OutputVector& new_args)
 }
 
 void Load::set_output_port_descriptor(const MemoryAccess::PortDescriptor& desc, const size_t i) {
-    // Load is one-port MemoryAccess operation. To simulate this behavior input_desc = output_desc
-    set_input_port_descriptor(desc, i);
+    throw ov::Exception("Load node doesn't have memory access output port");
 }
 
 const MemoryAccess::PortDescriptor& Load::get_output_port_descriptor(const size_t i) const {
-    // Load is one-port MemoryAccess operation. To simulate this behavior input_desc = output_desc
-    return get_input_port_descriptor(i);
+    throw ov::Exception("Load node doesn't have memory access output port");
```

Comment:
1) You don't have to override `get_output_port_descriptor` now, since the MemoryAcess implementation will throw [anyway](https://github.com/openvinotoolkit/openvino/blob/6b80c5de66f042f7d414ed03748c75d7dfdbd97a/src/common/snippets/src/op/memory_access.cpp#L53), as far as we don't have and output port desc.
2) The same is true for `Store::get_input_port_desc`
3) We still have one vulnerability in this approach: one can dynamic_cast Load to MemoryAccess and create (invalid) out port descriptor by calling `MemoryAcess::validate_and_infer_types()`. Therefore I suggest to remove `m_input_ports` [resize](https://github.com/openvinotoolkit/openvino/blob/6b80c5de66f042f7d414ed03748c75d7dfdbd97a/src/common/snippets/src/op/memory_access.cpp#L18-L23) from there, it should happen in a child class constructor, not here.

Reply:
1. Oh, really, thanks!
2. Done
3. Right. I have added parameters `input_count` and `output_coun` for `MemoryAcccess` ctor to avoid code duplication in general. But anyway we can manually set them in derived class ctors. Thanks!